plugins/stub_resolver/stub_resolver_main.ml

```diff
@@ -0,0 +1,110 @@
+open Core_kernel
+open Bap.Std
+open Bap_core_theory
+open Bap_knowledge
+include Self ()
+
+let is_section name v =
+  match Value.get Image.section v with
+  | Some x -> String.(x = name)
+  | _ -> false
+
+let eval slot exp =
+  let cls = Knowledge.Slot.cls slot in
+  match Knowledge.run cls exp (Toplevel.current ()) with
+  | Ok (v,_) -> Ok (Knowledge.Value.get slot v)
+  | Error conflict -> Error conflict
+
+let find_aliases (tid : tid) =
+  match eval Theory.Label.aliases (Knowledge.return tid) with
+  | Ok names -> names
+  | Error _ -> Set.empty (module String)
+
+let section_memory proj sec_name =
+  let collect_addresses addrs (mem,_ )=
+    Memory.foldi mem ~word_size:`r8 ~init:addrs
+      ~f:(fun addr _ acc -> Set.add acc addr) in
+  Memmap.filter (Project.memory proj) ~f:(is_section sec_name) |>
+  Memmap.to_sequence |>
+  Seq.fold ~init:(Set.empty (module Addr)) ~f:collect_addresses
+
+
+module Plt_resolver = struct
+
+  let memory_contains m sub =
+    match Term.get_attr sub address with
+    | None -> false
+    | Some addr -> Set.mem m addr
+
+  (* pre: all functions have different names
+     pre: there aren't any stubs with the same alias *)
```

Comment:
the second precondition is not holding, devise an algorithm that doesn't require this.